src/App.tsx

```diff
@@ -1,24 +1,365 @@
+        setErrorMessage('Unable to delete a todo');
+        throw err;
+      })
+      .finally(() => {
+        setTempTodos(null);
+        setIsLoad(false);
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+        myInputRef.current?.focus();
+      });
+  };
+
+  function filterTodos() {
+    switch (statusTodo) {
+      case Status.Active:
+        return todos.filter(todo => !todo.completed);
+      case Status.Completed:
+        return todos.filter(todo => todo.completed);
+      case Status.All:
+        return todos;
+      default: return todos;
+    }
+  }
+
+  useEffect(() => {
+    setErrorMessage('');
+    todoService.getTodos(USER_ID)
+      .then((response: SetStateAction<Todo[]>) => setTodos(response))
+      .catch(() => setErrorMessage('Unable to load todos'))
+      .finally(() => setInterval(() => setErrorMessage(''), 3000));
+  }, []);
+
+  function addTodo({
+    title,
+    userId,
+    completed,
+  }: Todo) {
+    setErrorMessage('');
+
+    const newTamperTodo = {
+      title,
+      userId,
+      completed,
+      id: 0,
+    };
+
+    if (!title.trim()) {
+      setErrorMessage('Title should not be empty');
+      setTimeout(() => {
+        myInputRef.current?.focus();
+      }, 0);
+
+      return Promise.resolve();
+    }
+
+    setTempTodo(newTamperTodo);
+
+    return todoService.addTodos({
+      title,
+      userId,
+      completed,
+    })
+      .then(
+        (newTodo: Todo) => {
+          if (newTodo.title) {
+            setTodos(currentTodos => [...currentTodos, newTodo]);
+          }
+        },
+      )
+      .catch((err: unknown) => {
+        setErrorMessage('Unable to add a todo');
+        setTimeout(() => {
+          myInputRef.current?.focus();
+        }, 0);
+        throw err;
+      })
+      .finally(() => {
+        setInterval(() => setErrorMessage(''), 3000);
+        setTempTodo(null);
+      });
+  }
+
+  async function deleteTodo(id: number) {
+    setIsLoad(true);
+    const newTamperTodo = todos.find(todo => todo.id === id);
+
+    if (newTamperTodo) {
+      setSelectedTodo(newTamperTodo);
+    }
+
+    return todoService.deleteTodos(id)
+      .then(() => {
+        setTodos(currentPosts => currentPosts.filter(todo => todo.id !== id));
+        setTimeout(() => {
+          myInputRef.current?.focus();
+        }, 0);
+      })
+      .catch((err) => {
+        setTodos(todos);
+        setErrorMessage('Unable to delete a todo');
+
+        throw err;
+      })
+      .finally(() => {
+        setIsLoad(false);
+        setInterval(() => setErrorMessage(''), 3000);
+      });
+  }
+
+  async function updateTitleTodo(updateTodo: Todo) {
+    setSelectedTodo(updateTodo);
+    const isSameTitle = todos.some(todo => todo.title === updateTodo.title);
+
+    if (!updateTodo.title.trim()) {
+      // eslint-disable-next-line no-useless-catch
+      try {
+        await deleteTodo(updateTodo.id);
+      } catch (error) {
+        throw error;
```

Comment:
```suggestion
        setErrorMessage('unable to update todo')
```

Reply:
I need here 'throw error' to pass it to section and make other adjustments. setErrorMessage work anyway